File: RefactoringPracticeI/GameInventory/ApprovalTest/ApprovalTest.cs
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using System.Text;

namespace RefactoringExerciseI.ApprovalTest
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ApprovalTest
    {
        [Test]
        public void ThirtyDays()
        {
            StringBuilder fakeoutput = new();
            Console.SetOut(new StringWriter(fakeoutput));
            Console.SetIn(new StringReader("a\n"));

            Program.Main();
            string output = fakeoutput.ToString();

            Approvals.Verify(output);
        }
    }
}

File: RefactoringPracticeI/src/test/java/approval/ApprovalTest.java
package approval;

import org.assuresoft.Program;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import static org.junit.jupiter.api.Assertions.assertNotNull;

public class ApprovalTest {
  @Test
  public void thirtyDays() {
    ByteArrayOutputStream fakeOutput = new ByteArrayOutputStream();
    System.setOut(new PrintStream(fakeOutput));
    System.setIn(System.in);

    Program.main(new String[0]);

    String output = fakeOutput.toString();

    assertNotNull(output);
  }
}
